app/controllers/ems_common.rb

```diff
@@ -832,6 +871,40 @@ def get_form_vars
     set_verify_status
   end
 
+  def set_record_creds(ems)
+    creds = {}
+    creds[:default] = {
+      :userid   => @edit[:new][:default_userid],
```

Comment:
I'd start with this one to see if you can do it in isolation... if it works well so you can now test that method without doing bloody `instance_variable_set`, then you know the recipe and can do the other one later.